### PR TITLE
Infra changes for Microsoft.WinGet.Client and AppInstallerCLIE2ETests

### PR DESCRIPTION
**Pull request by msftrubengu**
This PR contains several infra changes:

- Use same stylecops file for Microsoft.WinGet.Client and AppInstallerCLIE2ETests. In the future we should point the same stylecops file for different csproj in this repository.
- Mark warnings as errors in Microsoft.WinGet.Client for release builds.
- Because of that, a lot of AppInstallerCLIE2ETests. All of them are because of missing comments, edit the header and follow stylecops standards on where to place things within a class, etc.
- Create Create-CrescendoFunctions.ps1 to easily create Microsoft.WinGet.Client.psm1 in the future if necessary. It can sadly be fully automated because of the dynamic expression in Microsoft.WinGet.Client.psd1. Not really needed, but nice to have.
- Move PowerShell module output to outdir\PowerShell\Microsoft.WinGet.Client. Every directory under the PowerShell output directory will be a different WinGet module.
- Move Microsoft.WinGet.Client PowerShell specific files to a Modules directory. This helps us throw everything there without needing to edit the after targets copy files task with each specific file.
- Create Initialize-LocalWinGetModules.ps1 as a helper to initialize WinGet's PowerShell modules locally. Used for dev testing.
- Update-BinVer fails in DevOps agent if git describe fails even though the script is supposed to set the version as 0. I believe is something in the output of git describe that makes the task fail, but I didn't want to go that deep, specially because of the fallback. The other option was to modify the pipeline file to don't execute for manual triggers, but I don't want us to block that possibility. This change allows anyone (mostly) to create new pipeline from their fork and run the entire build with all the tests without needing to create a PR or a topic branch in the repo. It would be nice if at run time it detects the secure files are not there and not immediately failing then don't run the tests that require those files, but that seems to be a DevOps constrain (don't even trigger the pipeline if the files don't exist). 
- Modify pipeline file to remove unnecessary calls to scripts. Now that Visual Studio copies the necessary files at build time, there's no need to run the scripts.

In preparation of https://github.com/microsoft/winget-cli/issues/2697

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-cli/pull/2746)

### CODE REVIEW

**Review by ryfu-msft — APPROVED** [review]

File: src/PowerShell/Microsoft.WinGet.Client/Microsoft.WinGet.Client.csproj [hunk 1]
```diff
@@ -5,23 +5,28 @@
...
+    <DocumentationFile>$(OutputPath)\Microsoft.WinGet.Client.xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(WingetUseProdClsids)' == 'true'">
+    <DefineConstants>USE_PROD_CLSIDS</DefineConstants>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(WingetUseProdClsids)' == 'true'">
     <DefineConstants>USE_PROD_CLSIDS</DefineConstants>
   </PropertyGroup>
```
Comment (yao-msft) [inline src/PowerShell/Microsoft.WinGet.Client/Microsoft.WinGet.Client.csproj:26]: nit: this is duplicate

File: src/PowerShell/Microsoft.WinGet.Client/Microsoft.WinGet.Client.csproj [hunk 2]
```diff
@@ -44,78 +49,90 @@
...
-		<PackageReference Include="Microsoft.Windows.CsWinRT" Version="1.6.5" />
-	</ItemGroup>
+  <PropertyGroup>
+    <!-- Disable warning for CS1591 because auto generated Microsoft.Management.Deployment -->
+    <!-- For our code, we still have stylecops acting on it. -->
+    <NoWarn>1591</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)'=='Release'">
+    <WarningsAsErrors />
```
Comment (yao-msft) [inline src/PowerShell/Microsoft.WinGet.Client/Microsoft.WinGet.Client.csproj:59]: ><WarningsAsErrors /> [](http://example.com/codeflow?start=4&length=20)

nit: remove

File: src/stylecop.json [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
   "$schema": "https://raw.githubusercontent.com/DotNetAnalyzers/StyleCopAnalyzers/master/StyleCop.Analyzers/StyleCop.Analyzers/Settings/stylecop.schema.json",
   "settings": {
+    "maintainabilityRules": {
+      "topLevelTypes": [ "class", "interface" ]
+    },
     "documentationRules": {
       "companyName": "Microsoft Corporation",
       "copyrightText": "    Copyright (c) {companyName}. Licensed under the MIT License.",
-      "headerDecoration": "-----------------------------------------------------------------------------"
+      "headerDecoration": "-----------------------------------------------------------------------------",
```
Comment (yao-msft) [inline src/stylecop.json:10]: >, [](http://example.com/codeflow?start=105&length=1)

nit: from json perspective, this change is actually incorrect

**Review by yao-msft — APPROVED** [review]
:shipit:

**Pull request merged**